sdk/historian/src/main/java/org/dsa/iot/historian/database/Watch.java

```diff
@@ -28,25 +39,27 @@
     private Node realTimeValue;
     private String watchedPath;
 
-    private Node lastWrittenValue;
     private Node startDate;
     private Node endDate;
     private boolean enabled;
 
     // Values that must be handled before the buffer queue
     private long lastWrittenTime;
+
+    // Used to represent last value to the database
+    private Node lastWrittenValue;
+
+    // Used for POINT_CHANGE
     private Value lastValue;
 
     public WatchUpdate getLastWatchUpdate() {
-        if (lastWatchUpdate == null) {
```

Comment:
That's a tough state machine to follow.  It looks like this change ensures that a new WatchUpdate is created for each interval instead of re-using the last one.  Was the issue that intervals simply had the incorrect time stamp?